content/api/commands/mount.md

```diff
@@ -0,0 +1,226 @@
+})
+```
+
+:::
+
+See [examples](#Examples) below for practical use cases.
+
+## Adding TypeScript Typings for `cy.mount` Commands
+
+When working in
+[TypeScript](https://docs.cypress.io/guides/tooling/typescript-support), you
+will need to add some custom typings for your commands to get code completion
+and avoid any TypeScript errors. Below are the typings for the above examples
+for React and Vue:
+
+:::react-vue-example
+
+```ts
+// cypress.d.ts
+import { MountOptions, MountReturn } from '@cypress/react'
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Mounts a React node
+       * @param jsx React Node to mount
+       * @param options Additional options to pass into mount
+       */
+      mount(
+        jsx: React.ReactNode,
+        options?: MountOptions
+      ): Cypress.Chainable<MountReturn>
+    }
+  }
+}
+```
+
+```ts
+import { mount } from '@cypress/vue'
+type MountParams = Parameters<typeof mount>
+
+declare global {
+  namespace Cypress {
+    type ComponentParam = MountParams[0] | JSX.Element
+    type OptionsParam = MountParams[1]
+    interface Chainable {
+      /**
+       * Helper mount function for Vue Components
+       * @param component Vue Component or JSX Element to mount
+       * @param options Options passed to Vue Test Utils
+       */
+      mount(component: ComponentParam, options?: OptionsParam): Chainable<any>
+    }
+  }
+}
+```
+
+:::
+
+The typings will need to be in a location that any code can access in the
+application. We recommend adding them to a `cypress.d.ts` file at the root of
+the application and updating any tsconfig.json files to reference the typings
+file in the `include` option:
+
+```json
+"include": ["./src", "cypress.d.ts"]
+```
+
+## Additional Mount Commands
+
+You can create additional mount commands to fit your needs. For instance, there
+might be times when you need to do additional setup for certain tests that other
+tests don't need. In this case, it is good to create a new command for those
+tests.
+
+Below are some examples for specific use cases.
+
+## React Examples
+
+### React Router
+
+If you have a component that consumes a hook or component from React Router, you
+will need to make sure the component has access to a React Router provider.
+Below is a sample mount command that uses `MemoryRouter` to wrap the component.
+Setup props for `MemoryRouter` can be passed in the options param as well:
+
+```jsx
+import { mount } from '@cypress/react'
+import { MemoryRouter } from 'react-router-dom'
+
+Cypress.Commands.add('mountWithRouter', (component, options) => {
+  const { routerProps = { initialEntries: ['/'] }, ...mountOptions } = options
+
+  const wrapped = <MemoryRouter {...routerProps}>{component}</MemoryRouter>
+
+  return mount(wrapped, mountOptions)
+})
+```
+
+You can pass in custom props for the `MemoryRouter` in the `options` param.
+
+Example usage:
+
+```jsx
+it('Logout link should appear when logged in', () => {
```

Comment:
Can you add an example where it uses the default `routerProps` value? Also, it might be helpful to show a test asserting the the specified `initialEntries` value is correct. Something like:

```jsx
it('should set the home page link to be active by default', () => {
  cy.mountWithRouter(<LeftNav />)
  cy.contains('Home').should('have.class', 'active')
})

it('should set the about page link to be active on the about page', () => {
  cy.mountWithRouter(<LeftNav />, {
    routerProps: {
      initialEntries: ['/about'],
    },
  })
  cy.contains('About').should('have.class', 'active')
})
```